### DISCUSSION TITLE
Silent Install Process Hacker with driver + taskmgr replacement

### DISCUSSION THREAD

**Discussion by teknixstuff** [winsiderss/systeminformer#1957] [Questions / Answers]
What arguments are required to install process hacker 2.39 with the driver and taskmgr replacement enabled?

**Comment by Advait251206** [comment]
For **Process Hacker 2.39**, the installer is based on Inno Setup and supports silent installation switches such as:

```cmd
processhacker-2.39-setup.exe /VERYSILENT /SUPPRESSMSGBOXES /NORESTART
```

:contentReference[oaicite:0]{index=0}

To enable both the **KProcessHacker driver** and **Task Manager replacement** during installation, the commonly documented tasks are:

```cmd
processhacker-2.39-setup.exe /VERYSILENT /NORESTART /MERGETASKS=create_kph_service,set_default_taskmgr
```

Where:

- `create_kph_service` = installs the KProcessHacker driver/service.
- `set_default_taskmgr` = configures Process Hacker as the Task Manager replacement.

:contentReference[oaicite:1]{index=1}

Alternatively, if the driver was not installed during setup, Process Hacker also exposes command-line options to install it later:

```cmd
ProcessHacker.exe -installkph
```

and remove it with:

```cmd
ProcessHacker.exe -uninstallkph
```

:contentReference[oaicite:2]{index=2}

So the command I'd try first is:

```cmd
processhacker-2.39-setup.exe /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /MERGETASKS=create_kph_service,set_default_taskmgr
```

If you're deploying this in an enterprise environment, I'd recommend testing on a VM first, since the available installer tasks can vary slightly between builds and forks of Process Hacker. :contentReference[oaicite:3]{index=3}
